src/classes/CRLP_Rollup_SEL.cls

```diff
@@ -178,43 +444,65 @@ public class CRLP_Rollup_SEL {
      * @return List<DescribeFieldResult>
      */
     public static List<Schema.DescribeFieldResult> getReferencedFieldsByObject(SObjectType objType) {
-        String objName = UTIL_Describe.getObjectDescribe(objType).getName();
-        Map<String, Schema.DescribeFieldResult> allFields = UTIL_Describe.getAllFieldsDescribe(objName);
-        Map<String, Schema.DescribeFieldResult> referencedFields = new Map<String, Schema.DescribeFieldResult>();
-        for (Rollup__mdt rollup : cachedRollups) {
-            if (rollup.Active__c == true) {
-                if (rollup.Summary_Object__r.QualifiedApiName == objName) {
-                    String fldName = rollup.Summary_Field__r.QualifiedApiName;
-                    referencedFields.put(fldName, allFields.get(fldName.toLowerCase()));
-                }
-                if (rollup.Detail_Object__r.QualifiedApiName == objName && rollup.Detail_Field__c != null) {
-                    String fldName = rollup.Detail_Field__r.QualifiedApiName;
-                    referencedFields.put(fldName, allFields.get(fldName.toLowerCase()));
-                }
-                if (rollup.Amount_Object__r.QualifiedApiName == objName && rollup.Amount_Field__r != null) {
-                    String fldName = rollup.Amount_Field__r.QualifiedApiName;
-                    referencedFields.put(fldName, allFields.get(fldName.toLowerCase()));
-                }
-                if (rollup.Date_Object__r.QualifiedApiName == objName && rollup.Date_Field__r != null) {
-                    String fldName = rollup.Date_Field__r.QualifiedApiName;
-                    referencedFields.put(fldName, allFields.get(fldName.toLowerCase()));
-                }
-            }
+        // Collect SObjectFields in set in case there are duplicates
+        final Set<SObjectField> fields = new Set<SObjectField>();
+
+        // Always add the AccountId and Primary_Contact__c for the Opportunity object
```

Comment:
I moved around the order of adding return values (`SObjectField` --> `DescribeFieldResult`) to make it easier building test data.  The order of returned `DescribeFieldResult` doesn't matter to production code.

Old Order
----
1. Rollup Fields
2. `CurrencyIsoCode`
3. Special Opportunity Fields

New Order
----
1. Special Opportunity Fields
2. `CurrencyIsoCode`
3. Rollup Fields